src/libcommon/utility/utility.cpp

```diff
@@ -235,28 +235,43 @@ QString CommonUtility::escape(const QString &in) {
 }
 
 bool CommonUtility::stringToAppStateValue(const std::string &stringFrom, AppStateValue &appStateValueTo) {
+    bool res = true;
+    std::string appStateValueType = "Unknown";
     if (std::holds_alternative<std::string>(appStateValueTo)) {
         appStateValueTo = stringFrom;
+        appStateValueType = "std::string";
     } else if (std::holds_alternative<int>(appStateValueTo)) {
+        appStateValueType = "int";
         try {
             appStateValueTo = std::stoi(stringFrom);
         } catch (const std::invalid_argument &) {
-            return false;
+            res = false;
         }
     } else if (std::holds_alternative<LogUploadState>(appStateValueTo)) {
+        appStateValueType = "LogUploadState";
         try {
             appStateValueTo = static_cast<LogUploadState>(std::stoi(stringFrom));
         } catch (const std::invalid_argument &) {
-            return false;
+            res = false;
         }
     } else if (std::holds_alternative<int64_t>(appStateValueTo)) {
+        appStateValueType = "int64_t";
```

Comment:
I would create a utility fonction that return the string (based of the std::variant index) instead of hardcoding those string here. For example see 'Utility::logLevel2Str'